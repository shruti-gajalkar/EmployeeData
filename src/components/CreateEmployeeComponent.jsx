import React, { ReactDOM, useState, useEffect } from 'react';
import EmployeeService from '../services/EmployeeService';
import Modal from 'react-modal';
if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#root');

function CreateEmployeeComponent(props) {

    const customStyles = {
        content: {
            top: '50%',
            left: '50%',
            right: 'auto',
            bottom: 'auto',
            marginRight: '-20%',
            width: '50%',
            transform: 'translate(-50%, -50%)',
        },
    };

    const [state, setState] = useState({});

    useEffect(() => {
        if (state && state.id === '_add') {
            return
        }
    }, []);

    const saveOrUpdateEmployee = (e) => {
        e.preventDefault();
        let employee = { firstName: state.firstName, lastName: state.lastName, emailId: state.emailId };
        console.log('employee => ' + JSON.stringify(employee));

        EmployeeService.createEmployee(employee).then(res => {
            props.setShow(false);
            //props.history.push('/employees');
            window.location.reload();
        });

    }

    const handleOnChange = (name, event) => {
        event.preventDefault();
        const field = {}
        field[name] = event.target.value
        setState({ ...state, ...field })
    }

    const cancel = () => {
        props.setShow(false);
        //props.history.push('/employees');
    }

    return (
        <div className>
            <Modal
                isOpen={props.show}
                //onAfterOpen={afterOpenModal}
                onRequestClose={cancel}
                style={customStyles}
                contentLabel="Example Modal"
            >

                <div className="row">
                    <div className="col-md-6 offset-md-3 offset-md-3">
                        <h3 className="text-center">Add Employee</h3>
                        <div className="card-body">
                            <form>
                                <div className="form-group">
                                    <label> First Name: </label>
                                    <input placeholder="First Name" name="firstName" className="form-control"
                                        onChange={(e) => handleOnChange('firstName', e)} />
                                </div>
                                <div className="form-group">
                                    <label> Last Name: </label>
                                    <input placeholder="Last Name" name="lastName" className="form-control"
                                        onChange={(e) => handleOnChange('lastName', e)} />
                                </div>
                                <div className="form-group">
                                    <label> Email Id: </label>
                                    <input placeholder="Email Address" name="emailId" className="form-control"
                                        onChange={(e) => handleOnChange('emailId', e)} />
                                </div>

                                <button className="btn btn-success" onClick={(e) => saveOrUpdateEmployee(e)}>Save</button>
                                <button className="btn btn-danger" onClick={() => cancel()} style={{ marginLeft: "10px" }}>Cancel</button>
                            </form>
                        </div>
                    </div>
                </div>

            </Modal>
        </div>
    )
}

export default CreateEmployeeComponent
