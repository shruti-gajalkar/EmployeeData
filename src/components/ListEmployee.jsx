import React, { useState, useEffect } from 'react';
import EmployeeService from '../services/EmployeeService'
import CreateEmployeeComponent from './CreateEmployeeComponent';
import EmployeeRecords from './EmployeeRecords'
import Pagination from './Pagination';

const ListEmployee = (props) => {

    const [posts, setPosts] = useState([]);
    const [loading, setLoading] = useState(false);
    const [currentPage, setCurrentPage] = useState(1);
    const [postsPerPage] = useState(10);
    const [show, setShow] = useState(false);
    useEffect(() => {
        EmployeeService.getEmployees().then((res) => {
            setLoading(true);
            setPosts(res.data);
            setLoading(false);
        });
    }, []);

    // Get current posts
    const indexOfLastPost = currentPage * postsPerPage;
    const indexOfFirstPost = indexOfLastPost - postsPerPage;
    const currentPosts = posts.slice(indexOfFirstPost, indexOfLastPost);

    // Change page
    const paginate = pageNumber => setCurrentPage(pageNumber);

    const addEmployee = () => {
        setShow(true);
    }


    return (
        <div>
            <h2 className="text-center">Employees List</h2>
            <div className="row">
                <button className="btn btn-primary" onClick={()=>addEmployee()}> Add Employee</button>
            </div>
            <br></br>
            <EmployeeRecords 
            employees={currentPosts} 
            loading={loading} />
            <Pagination
                postsPerPage={postsPerPage}
                totalPosts={posts.length}
                paginate={paginate}
            />
            {show && <CreateEmployeeComponent
            show={show}
            setShow={setShow}
            history={props.history}
            />} 
        </div>
    )
}


export default ListEmployee
