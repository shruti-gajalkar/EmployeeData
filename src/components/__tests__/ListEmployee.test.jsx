import React from 'react';
import ReactDOM from 'react-dom';

import Enzyme, { shallow, render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() })
import ListEmployee from '../ListEmployee.jsx';
import CreateEmployeeComponent from '../CreateEmployeeComponent'
import HeaderComponent from '../HeaderComponent.js';


it('renders correctly enzyme', () => {
    const wrapper = shallow(<ListEmployee />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders correctly CreateEmployeeComponent enzyme', () => {
    const wrapper = shallow(<CreateEmployeeComponent />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders correctly HeaderComponent enzyme', () => {
    const wrapper = shallow(<HeaderComponent />)
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('the default value for  fields should be empty', () => {
    const wrapper = shallow(<CreateEmployeeComponent />);
    expect(wrapper.find('input[name="emailId"]').exists()).toBe(true);
  });
  it('the default value for  fields should be empty', () => {
    const wrapper = shallow(<CreateEmployeeComponent />);
    expect(wrapper.find('input[name="lastName"]').exists()).toBe(true);
  });
  it('the default value for  fields should be empty', () => {
    const wrapper = shallow(<CreateEmployeeComponent />);
    expect(wrapper.find('input[name="firstName"]').exists()).toBe(true);
  });

