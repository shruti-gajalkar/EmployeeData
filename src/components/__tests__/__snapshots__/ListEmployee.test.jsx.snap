// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly CreateEmployeeComponent enzyme 1`] = `
<div
  className=""
>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentElement={[Function]}
    contentLabel="Example Modal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-20%",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "50%",
        },
      }
    }
  >
    <div
      className="row"
    >
      <div
        className=" col-md-6 offset-md-3 offset-md-3"
      >
        <h3
          className="text-center"
        >
          Add Employee
        </h3>
        <div
          className="card-body"
        >
          <form>
            <div
              className="form-group"
            >
              <label>
                 First Name: 
              </label>
              <input
                className="form-control"
                name="firstName"
                onChange={[Function]}
                placeholder="First Name"
              />
            </div>
            <div
              className="form-group"
            >
              <label>
                 Last Name: 
              </label>
              <input
                className="form-control"
                name="lastName"
                onChange={[Function]}
                placeholder="Last Name"
              />
            </div>
            <div
              className="form-group"
            >
              <label>
                 Email Id: 
              </label>
              <input
                className="form-control"
                name="emailId"
                onChange={[Function]}
                placeholder="Email Address"
              />
            </div>
            <button
              className="btn btn-success"
              onClick={[Function]}
            >
              Save
            </button>
            <button
              className="btn btn-danger"
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              Cancel
            </button>
          </form>
        </div>
      </div>
    </div>
  </Modal>
</div>
`;

exports[`renders correctly CreateEmployeeComponent enzyme 2`] = `
<div
  className=""
>
  <Modal
    ariaHideApp={true}
    bodyOpenClassName="ReactModal__Body--open"
    closeTimeoutMS={0}
    contentElement={[Function]}
    contentLabel="Example Modal"
    isOpen={false}
    onRequestClose={[Function]}
    overlayElement={[Function]}
    parentSelector={[Function]}
    portalClassName="ReactModalPortal"
    preventScroll={false}
    role="dialog"
    shouldCloseOnEsc={true}
    shouldCloseOnOverlayClick={true}
    shouldFocusAfterRender={true}
    shouldReturnFocusAfterClose={true}
    style={
      Object {
        "content": Object {
          "bottom": "auto",
          "left": "50%",
          "marginRight": "-20%",
          "right": "auto",
          "top": "50%",
          "transform": "translate(-50%, -50%)",
          "width": "50%",
        },
      }
    }
  >
    <div
      className="row"
    >
      <div
        className=" col-md-6 offset-md-3 offset-md-3"
      >
        <h3
          className="text-center"
        >
          Add Employee
        </h3>
        <div
          className="card-body"
        >
          <form>
            <div
              className="form-group"
            >
              <label>
                 First Name: 
              </label>
              <input
                className="form-control"
                name="firstName"
                onChange={[Function]}
                placeholder="First Name"
              />
            </div>
            <div
              className="form-group"
            >
              <label>
                 Last Name: 
              </label>
              <input
                className="form-control"
                name="lastName"
                onChange={[Function]}
                placeholder="Last Name"
              />
            </div>
            <div
              className="form-group"
            >
              <label>
                 Email Id: 
              </label>
              <input
                className="form-control"
                name="emailId"
                onChange={[Function]}
                placeholder="Email Address"
              />
            </div>
            <button
              className="btn btn-success"
              onClick={[Function]}
            >
              Save
            </button>
            <button
              className="btn btn-danger"
              onClick={[Function]}
              style={
                Object {
                  "marginLeft": "10px",
                }
              }
            >
              Cancel
            </button>
          </form>
        </div>
      </div>
    </div>
  </Modal>
</div>
`;

exports[`renders correctly HeaderComponent enzyme 1`] = `
<div>
  <header>
    <nav
      className="navbar navbar-expand-md navbar-dark bg-dark"
    >
      <div>
        <a
          className="navbar-brand"
        >
          Employee Management App
        </a>
      </div>
    </nav>
  </header>
</div>
`;

exports[`renders correctly enzyme 1`] = `
<div>
  <h2
    className="text-center"
  >
    Employees List
  </h2>
  <div
    className="row"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
       Add Employee
    </button>
  </div>
  <br />
  <EmployeeRecords
    employees={Array []}
    loading={false}
  />
  <Pagination
    paginate={[Function]}
    postsPerPage={10}
    totalPosts={0}
  />
</div>
`;

exports[`renders correctly enzyme 2`] = `
<div>
  <h2
    className="text-center"
  >
    Employees List
  </h2>
  <div
    className="row"
  >
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
       Add Employee
    </button>
  </div>
  <br />
  <EmployeeRecords
    employees={Array []}
    loading={false}
  />
  <Pagination
    paginate={[Function]}
    postsPerPage={10}
    totalPosts={0}
  />
</div>
`;
